File: JAVA-Basic/JavaBasic/src/WhileLoop/Exercise/Walking.java
package WhileLoop.Exercise;

import java.util.Scanner;

public class Walking {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int stepsCounter = 0;

        while (stepsCounter < 10000) {
            String currentCheck = scan.nextLine();

            if (currentCheck.equals("Going home")) {
                stepsCounter += Integer.parseInt(scan.nextLine());
                break;
            } else {
                int currentSteps = Integer.parseInt(currentCheck);
                stepsCounter += currentSteps;
            }
        }
        int difference = Math.abs(10000 - stepsCounter);
        if (stepsCounter >= 10000) {
            System.out.println("Goal reached! Good job!");
            System.out.printf("%d steps over the goal!", difference);
        } else {
            System.out.printf("%d more steps to reach goal.", difference);
        }
    }
}

File: 01.JS-Basic/WHILE-ЦИКЪЛ Упражнение/Walking.js
function walking(input) {

    let goal = 10000
    let i = 0
    let score = 0

    while (score <= goal) {

        let dailySteps = input[i]

        if (dailySteps === "Going home") {
            score += Number(input[i + 1]);
            break;
        } else {
            score += Number(dailySteps)
        }
        i++
    }
    if (score >= goal) {
        console.log("Goal reached! Good job!")
        console.log(`${score - goal} steps over the goal!`)
    } else {
        console.log(`${goal - score} more steps to reach goal.`)
    }
}
walking(["1500",
    "3000",
    "250",
    "1548",
    "2000",
    "Going home",
    "2000"])

